app/controllers/admin/claims_controller.rb

```diff
@@ -3,13 +3,22 @@ class Admin::ClaimsController < Admin::BaseAdminController
 
   def index
     claims = Claim.includes(eligibility: [:claim_school, :current_school]).submitted.order(:submitted_at)
-    csv = TslrClaimsCsv.new(claims)
+    csv = StudentLoans::ClaimsCsv.new(claims)
 
     respond_to do |format|
       format.csv { send_file csv.file, type: "text/csv", filename: "claims.csv" }
     end
   end
 
+  def payroll
```

Comment:
At the risk of being another person throwing my oar in here, this doesn't feel very RESTful, can we maybe control the type of csv being returned via a GET parameter? Especially given there's a lot of repeated code in this action too

Reply:
I've implemented the solution you suggested. It's worth mentioning the amount of code those two action share will reduce once were using approved only claims in the payroll CSV but I guess we can cross that bridge when we get to it

Reply:
I'm less convinced by this change tbh. The repetition was in only two places and was boring, standard rails code, which we've replaced with a helper method containing conditional logic and "magic" string constants. Before, the routing helpers gave us concrete methods that would fail hard if there was a typo in the url helper calls to generate these links. Now they'll happily generate a link to an action that will respond with nothing if the magic string doesn't match the ones in the conditional logic. And as @robbiepaul points out, the two CSV files won't include the same claims: payroll is only meant to include "approved" claims, whilst the other is for "submitted" claims.

Generally, any attempt to DRY up code and introduce abstractions should be done with care as there is a cost in introducing new abstractions and entangling two bits of loguc. I stick to the rule of thumb that unless it is blindingly and obviously the case that the code is 100% identical, then I want to see three instances of duplication before I think about generalising the code to DRY it up.